src/webnn_native/ops/Transpose.cpp

```diff
@@ -21,15 +21,31 @@
 
 namespace webnn_native { namespace op {
 
-    MaybeError Transpose::Validate() {
-        MaybeError maybeError = OperatorBase::Validate();
+    MaybeError Transpose::CalculateShape() {
+        auto inputShape = mInputs[0]->Shape();
+        size_t rank = inputShape.size();
+        std::vector<int32_t> outputShape(rank);
+        for (size_t i = 0; i < rank; i++) {
+            outputShape[i] = inputShape[mPermutation[i]];
+        }
+        mOutputs[0]->SetShape(std::move(outputShape));
+        return {};
+    }
+
+    MaybeError Transpose::ValidateAndInferOutputInfo() {
+        MaybeError maybeError = OperatorBase::ValidateAndInferOutputInfo();
         if (maybeError.IsError()) {
             return maybeError;
         }
 
-        // the number of values in the sequence must be the same as the rank of the input tensor
-        size_t inputRank = mInputs[0]->Rank();
-        if (mPermutation.size() != size_t(inputRank)) {
+        auto inputShape = mInputs[0]->Shape();
+        if (inputShape.empty()) {
```

Comment:
Move it to base Operator, you can override ValidateAndInferOutputInfo with a argument like allowEmptyShape if the input shape allow empty shape.

Reply:
As discussed wit you, I will remove the validation for empty input shape. Thanks! @fujunwei